File: frontend/src/types/tool.ts
// 版本记录
export interface ToolVersion {
  id: number
  tool_id: number
  version: number
  code?: string
  description?: string
  description_zh?: string
  change_summary?: string
  created_at: string
}

export interface Tool {
  id: number
  name: string
  description: string
  description_zh?: string
  code: string
  language: string
  version: number
  tool_type?: string
  is_builtin: boolean
  created_by: number | null
  created_at: string
  updated_at: string
}

export interface ToolBrief {
  id: number
  name: string
  description: string
  description_zh?: string
  language: string
  version: number
  tool_type?: string
  is_builtin: boolean
  sub_tool_count?: number
  created_at: string
}

export interface ToolCreate {
  name: string
  description: string
  description_zh?: string
  code: string
  language: string
}

export interface ToolUpdate {
  name?: string
  description?: string
  description_zh?: string
  code?: string
  language?: string
}

export interface ToolImportResult {
  created: number
  updated: number
  skipped: number
}

export interface ToolExportItem {
  name: string
  description: string
  description_zh?: string
  code: string
  language: string
  version: number
}

File: backend/app/schemas/tool.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ToolCreate(BaseModel):
    name: str
    description: str
    description_zh: Optional[str] = None
    code: str
    language: str = "python"


class ToolBrief(BaseModel):
    id: int
    name: str
    description: str
    description_zh: Optional[str] = None
    language: str
    version: int
    is_builtin: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ToolUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    description_zh: Optional[str] = None
    code: str | None = None
    language: str | None = None


class ToolImportItem(BaseModel):
    name: str
    description: str = ""
    description_zh: Optional[str] = None
    code: str
    language: str = "python"
    tool_type: Optional[str] = "atomic"


class ToolExportItem(BaseModel):
    name: str
    description: str
    description_zh: Optional[str] = None
    code: str
    language: str
    version: int
    tool_type: Optional[str] = "atomic"

    model_config = {"from_attributes": True}


class ToolResponse(ToolBrief):
    code: str
    created_by: int | None
    updated_at: datetime


class ToolVersionResponse(BaseModel):
    id: int
    tool_id: int
    version: int
    code: Optional[str] = None
    description: Optional[str] = None
    description_zh: Optional[str] = None
    change_summary: Optional[str] = None
    created_at: Optional[str] = None

    model_config = {"from_attributes": True}
